### PR TITLE
Sponsorship

### PR DESCRIPTION
**Pull request by yungalyx**
still working on templating and having editable modals

### CODE REVIEW

File: pages/sponsorship.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    border: 1px solid #000000;
+    border-radius: 3px;
+    padding: 1px 1px;
+    background-color: #FFFFFF;
+`
+
+const HeaderCard = styled.div`
+    padding: 26px 28px;
+    font-weight: bold;
+    font-family: Apercu Pro;
```
Comment (ianmah) [inline pages/sponsorship.js:31]: We can probably set the font family higher up so we don't have to redefine it for every component

File: pages/sponsorship.js [hunk 2]
```diff
@@ -18,17 +18,11 @@ const customStyles = {
...
 
 
-const TableDiv = styled.div`
-    border: 1px solid #000000;
-    border-radius: 3px;
-    padding: 1px 1px;
-    background-color: #FFFFFF;
-`
-
+// TODO: move font family higher, reduce need to redefine
```
Comment (ianmah) [inline pages/sponsorship.js:21]: let's just remove the font for now

**Comment by ianmah** [comment]
Table doesn't seem to be lining up properly
image

File: pages/sponsorship.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+       
+        
+    }
+
+    
+    handleEdit = (e) => {
+        e.preventDefault();
+        //TODO: pop-up window 
+        this.setState({showEditWindow: true})
+    };
```
Comment (ianmah) [inline pages/sponsorship.js:76]: A bit more modern and cleaner to use this syntax for your buttons
`<Button onClick={() => this.handleEdit()}>`

Then you can do 
```suggestion
    handleEdit = () => {
        //TODO: pop-up window 
        this.setState({showEditWindow: true})
    };
```

**Comment by yungalyx** [comment]
@ianmah styling within the table cards is up to devs, i think theyre a bit different for each page

**Comment by ianmah** [comment]
Shouldn't the content at least line up with it's label? I think that's needed for every table

File: pages/sponsorship.js [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+       
+    }   
+
+    handleCancel = () => { 
+        this.setState({showEditWindow:false});
+    }
+
+    // sends and adds it to the list
+    handleSave = (event) => {
+        // prevents refreshing of the page
```
Comment (ianmah) [inline pages/sponsorship.js:99]: all of these functions can be updated to use the new syntax

**Comment by yungalyx** [comment]
@ianmah okok ill swap it over to maps, CMS figma made it seem like we could have multiple sponsors with same names 
figma

**Comment by yungalyx** [comment]
I only need to fix the fonts, timestamp format, and use correct images for the action buttons
+ the table dynamically configures to space items correctly based on browser size
yeet1


Sponsorship page updates correctly with CRUD firebase functionality in firebase.js 
yeet2

Waiting on modal component from Jenny's modal branch, but the form is completely working (with show and hide methods)
yeet3

**Comment by yungalyx** [comment]
I only need to fix the fonts, timestamp format, and use correct images for the action buttons
+ the table dynamically configures to space items correctly based on browser size
yeet1


Sponsorship page updates correctly with CRUD firebase functionality in firebase.js 
yeet2

Waiting on modal component from Jenny's modal branch, but the form is completely working (with show and hide methods)
yeet3

**Comment by Ciuffi** [comment]
Wow, looks amazing.

I'll take a look at the code tomorrow.

for now could you merge master into your branch so the CI can run?

**Comment by Ciuffi** [comment]
looks like you merged your local master. make sure to pull master before merging because this branch still has conflicts

File: utility/firebase.js [hunk 5]
```diff
@@ -199,43 +199,33 @@ const fireDb = {
...
-      const altRef = storage.ref(`${website}/alt${image}`);
-      await altRef.delete();
-    }
-    return true;
+  },
+  alexgetSponsors: async(website) => { 
+    var A = {};
+    const B = await db.collection("Hackathons").doc(website).collection("Sponsors").get();
+    B.docs.forEach(doc => {A[doc.id] = doc.data()})
+    return A;
```
Comment (Ciuffi) [inline utility/firebase.js:228]: A more functional way to do this would be like this:
```javascript
const refs = await db.collection("Hackathons").doc(website).collection("Sponsors").get();
const docs = refs.map(doc => doc.data())
return docs;
```

also you can just delete the old get function and rename this one to getSponsors

File: utility/firebase.js [hunk 6]
```diff
@@ -199,43 +199,33 @@ const fireDb = {
...
-    } catch (e) {
-      return false;
-    }
-    const ref = storage.ref(`${website}/${image}`);
+  deleteSponsor: async(website, sponsorName) => {
+    const ref = db
+      .collection("Hackathons")
+      .doc(website)
+      .collection("Sponsors")
+      .doc(sponsorName);
```
Comment (Ciuffi) [inline utility/firebase.js:221]: This delete function won't work properly because sponsors require an image upload (we host the images). I'm not sure if we're changing this but if not then we'll have to handle deleting the images from storage as well (like the original function did).

File: pages/sponsorship.js [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+    white-space: nowrap;    
+    overflow: hidden;         
+    text-overflow: ellipsis;
+`
+const Actions = styled.div`
+    padding-right: 80px;
+    flex: 1;
+`
+
+class SponsorshipPage extends React.Component {
```
Comment (Ciuffi) [inline pages/sponsorship.js:23]: I think we're using functional components for this project (it seems like the other PRs so far have been all functional components).

I know its a lot of work but we should stay consistent, could you switch this to a functional component?

File: pages/index.js [hunk 8]
```diff
@@ -1,13 +1,130 @@
...
+    line-height: 20px;
+`
+
 
 export default function Home() {
+
+  const googleSignIn = async (e) => {
+    e.preventDefault();
+    if (!firebase.apps.length) {
+      console.log(process.env.NEXT_PUBLIC_FIREBASE_API_KEY)
```
Comment (Ciuffi) [inline pages/index.js:97]: Home page should be on a different branch, not sure why its in this PR

File: components/modal.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+      if(!this.props.show) {
+          return null;
+      }
+    return <div>
+        <div>{this.props.children}</div>
+
+        </div>
+
+  }
+}
```
Comment (Ciuffi) [inline components/modal.js:14]: Again, ideally this would be a functional component.

File: components/card.js [hunk 10]
```diff
@@ -27,4 +27,23 @@ export const CardContent = styled.div`
...
+    padding: 1px 1px;
+    background-color: #FFFFFF;
+`
+
+
+export default ({children}) => (
+    <Card>
+        {children}
+    </Card>
+)
```
Comment (Ciuffi) [inline components/card.js:49]: are these card changes compatible with everyone else's use of the card component? 

Also card component changes should be in another PR

File: pages/sponsorship.js [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+                                        <button onClick={() => this.handleEdit(item.name)}> edit </button>
+                                    </Actions>
+                                </CardDiv>
+                            )}
+                        </div>
+                    </TableDiv>
+                    </CardContent>
+                </Card> 
+        
+                <form onSubmit={this.handleSave}>
```
Comment (Ciuffi) [inline pages/sponsorship.js:186]: is the hide/show for this form finnished yet?

Comment (yungalyx) [reply pages/sponsorship.js:187]: not yet! almost done

**Comment by yungalyx** [comment]
UPDATE SEP 5, 2020: 

1.  Corrected styling for table
sponsooo

2. changed Class component into functional component
- yeah this one kinda hurt 
exportdefauly


3. backend 
- simplified page creation just by specifying which hackathon event through `<SponsorshipPage name={EVENT} /> ` the routing in database uses the name to organize objects. 

backend

4. integrated + customized jenny's modal component
dior

There are some small styling issues, I had to use a custom createRef <input type='file'/>, so we will see quirks like being able to 'change' images or tiers in the VIEW modal. (but these changes don't actually carry through)


-- 

FEW MORE THINGS TO DO:
- [ ] add fonts
- [ ] fix date format
- [ ] make all input fields 'REQUIRED'.

**Comment by yungalyx** [comment]
o i also added sponsorship TIERS

**Review by Ciuffi — CHANGES_REQUESTED** [review]
This looks really great! Thanks for switching the component over to functional. I think it looks a lot cleaner that way.

A few things before we merge: Please move the sponsorship page into the [id] folder and set it up to work properly with the Page component. (You can properly just put everything inside sponsorship.js into [id]/sponsors.js). Make sure to grab the hackathon id from the props (you'll see this in sponsors.js and it'll look like this)
```javascript
export default ({ id, hackathons }) => (
  <Page currentPath={id} hackathons={hackathons}>
    Sponsors
  </Page>
);
```

Also, as we discussed at our last meeting, please also rename the sponsor.js file in the [id] folder to spocos.js so adblockers don't stop it from running lol.

Otherwise great work, seems like solid code and it looks great too. Oh, and make sure to fix the linting errors as well.

File: components/card.js [hunk 12]
```diff
@@ -27,4 +27,26 @@ export const CardContent = styled.div`
   padding: 40px;
 `;
 
-export default ({ children }) => <Card>{children}</Card>;
+// padding: top, right, bottom, left
+// jsutify contente space around
```
Comment (panjenny0) [inline components/card.js:31]:
```suggestion
```

File: components/card.js [hunk 13]
```diff
@@ -27,4 +27,26 @@ export const CardContent = styled.div`
...
+export const CardDiv = styled.div`
+    display: flex;
+    flex-flow: row nowrap;
+    justify-content: space-around;
+    align-items: center;
+    padding: ${props => props.padding};
+    border-bottom: 2px solid #EDEDED;
```
Comment (panjenny0) [inline components/card.js:38]:
```suggestion
    border-bottom: 2px solid ${COLOR.GRAY};
```

File: components/card.js [hunk 14]
```diff
@@ -27,4 +27,26 @@ export const CardContent = styled.div`
...
+`
+export const TableDiv = styled.div`
+    border: 1px solid #000000;
```
Comment (panjenny0) [inline components/card.js:41]:
```suggestion
    border: 1px solid ${COLOR.BLACK};
```

File: pages/sponsorship.js [hunk 15]
```diff
@@ -0,0 +1,301 @@
...
+
+    
+    useEffect(()=>{
+        loadFirebase();
+    }, []);
+
+    const loadFirebase = async() => {
+        const data = await fireDb.getSponsors(props.name);
+        setSponsors(data)
+        console.log(sponsors)
```
Comment (panjenny0) [inline pages/sponsorship.js:53]:
```suggestion
```

File: pages/sponsorship.js [hunk 16]
```diff
@@ -0,0 +1,301 @@
...
+    };
+    
+    
+    const handleEdit = (id) => {
+        //TODO: pop-up window
```
Comment (panjenny0) [inline pages/sponsorship.js:58]: Is this TODO still needed?
```suggestion
```

File: pages/sponsorship.js [hunk 17]
```diff
@@ -0,0 +1,301 @@
...
+
+    const handleView = (id) => {
+        setnewobj({
+            "name": sponsors[id].name,
+            "link": sponsors[id].link, 
+            "imgURL": sponsors[id].imgURL, 
+            "lastmod": sponsors[id].lastmod, 
+            "tier": sponsors[id].tier,
+        });
+        // TODO
```
Comment (panjenny0) [inline pages/sponsorship.js:78]:
```suggestion
```

File: pages/sponsorship.js [hunk 18]
```diff
@@ -0,0 +1,301 @@
...
+        name: "",
+        link: "",
+        imgURL: "", 
+        lastmod: "",
+        tier: ""
+    })
+    const [imgObject, setimgObject] = useState({});
+    const [imgFile, setimgFile] = useState({});
+    const [showEditWindow, setShowEditWindow] = useState(false);
+    const [modalAction, setmodalAction] = useState({});
```
Comment (panjenny0) [inline pages/sponsorship.js:43]: Nit: camelCase for consistency
```suggestion
    const [newobj, setNewObj] = useState({
        name: "",
        link: "",
        imgURL: "", 
        lastMod: "",
        tier: ""
    })
    const [imgObject, setImgObject] = useState({});
    const [imgFile, setImgFile] = useState({});
    const [showEditWindow, setShowEditWindow] = useState(false);
    const [modalAction, setModalAction] = useState({});
```

File: pages/sponsorship.js [hunk 19]
```diff
@@ -0,0 +1,301 @@
...
+        setSponsors(sponsors => ({
+            ...sponsors, 
+            [`${newobj.name}`]: newobj
+        }));
+
+        /*
+        this.state.sponsors[this.state.newobj.name] = this.state.newobj;
+        this.setState({ sponsors: this.state.sponsors });
+        */
+
```
Comment (panjenny0) [inline pages/sponsorship.js:138]:
```suggestion
```

**Review by panjenny0 — CHANGES_REQUESTED** [review]
Just some very minor things but otherwise looks good!
